meRg

FreshPenguin112 opened the pull request:
### Resolves

_What Github issue does this resolve (please include link)?_

- Resolves #

### Proposed Changes

_Describe what this Pull Request does_

### Reason for Changes

_Explain why these changes should be made_

### Test Coverage

_Please show how you have added tests to cover your changes_

### Browser Coverage
Check the OS/browser combinations tested (At least 2)

Mac
 * [ ] Chrome 
 * [ ] Firefox 
 * [ ] Safari
 
Windows
 * [ ] Chrome 
 * [ ] Firefox 
 * [ ] Edge
 
Chromebook
 * [ ] Chrome
 
iPad
* [ ] Safari

Android Tablet
* [ ] Chrome

GarboMuffin committed:
addons: Fix userstyles that use `@import`

GarboMuffin committed:
addons: Fix userstyle element class

GarboMuffin committed:
addons: Implement customCssVariables

GarboMuffin committed:
addons: Update to 1.29 - preliminary

GarboMuffin committed:
addons: Implement getCustomBlockColor and setCustomBlockColor

GarboMuffin committed:
addons: Update shared space selectors

GarboMuffin committed:
addons: Fix TW blocks

GarboMuffin committed:
addons: Implement customCssVariables `map`

GarboMuffin committed:
addons: Update to 1.29 - second pass

GarboMuffin committed:
addons: Import number-pad addon

GarboMuffin committed:
Merge pull request #671 from TurboWarp/addons-29

Addons 1.29

GarboMuffin committed:
addons: Mark number-pad as new

dependabot[bot] committed:
build(deps): bump scratch-vm from `ef871f7` to `1619bbf` (#667)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `ef871f7` to `1619bbf`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/ef871f794cdd660266b431c3d7878832e1c1fb2a...1619bbf795b19470bc91d4096a183a75cec2d14a)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump @turbowarp/scratch-l10n (#675)

Bumps [@turbowarp/scratch-l10n](https://github.com/TurboWarp/scratch-l10n) from 3.999.20221130194927 to 3.999.20221220224553.
- [Release notes](https://github.com/TurboWarp/scratch-l10n/releases)
- [Commits](https://github.com/TurboWarp/scratch-l10n/commits)

---
updated-dependencies:
- dependency-name: "@turbowarp/scratch-l10n"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Remove unshared project information from the homepage

Everyone who needs to see this has had over a month to see it.
The information still shows up when someone tries to view an unshared project.
Unshared projects were never the point. It's time to move on.

dependabot[bot] committed:
build(deps): bump scratch-vm from `1619bbf` to `e941bb0` (#672)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `1619bbf` to `e941bb0`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/1619bbf795b19470bc91d4096a183a75cec2d14a...e941bb0f04c8773c2d6bbff08725371613915159)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
addons: Implement paintEditorZoomControls shared space

GarboMuffin committed:
addons: Update assorted bug fixes from upstream

GarboMuffin committed:
addons: Add migration for editor-devtools split

GarboMuffin committed:
addons: fix most of https://github.com/TurboWarp/scratch-gui/issues/680

GarboMuffin committed:
addons: Rewrite userstyle loading

 - Breaking API change: [data-addon-id] is now [data-addons] and is a comma separated list instead of a single string.
 - Sheets included by multiple @ import statements are now deduplicated
 - Implement dynamic conditions for stylesheets.
 - editor-theme3's text color setting is dynamic again
Related to https://github.com/TurboWarp/scratch-gui/issues/680

GarboMuffin committed:
addons: editor-theme3 must have higher precedence than middle-click-popup and find-bar

Fixes https://github.com/TurboWarp/scratch-gui/issues/680

GarboMuffin committed:
addons: Update editor-theme3: fix error applying TW colors

GarboMuffin committed:
addons: Replace _twGetAsset with addon.tab.getResource

Instead of generating a _twGetAsset function for each JS file, we
import all of them in _runtime_entry.js. Upstream has expressed interest
in doing this as well.
pull.js has also been updated to warn about unwritten asset imports.

GarboMuffin committed:
Use dark flyout checkboxes in dark mode

GarboMuffin committed:
Don't show option to hide list outside of the editor

GarboMuffin committed:
addons: Apply scratch-addons-style class to stylesheets

Was regressed in previous stylesheet rewrite

GarboMuffin committed:
addons: Update addons

Updated to b07855ae3f7ddc231ae81b3d42a0e23a1a357b10
Upstream changed a lot of icons for some reason. There are also several
diff reduction changes in here.

GarboMuffin committed:
addons: Move 'if' parsing to SettingsStore

Addon API and settings page both had their own implementation. Now there
is one single implementation that is thoroughly tested.

GarboMuffin committed:
addons: Pull some updates

Notably more addon settings are dynamic than before.

GarboMuffin committed:
addons: Remove updateUserstylesOnSettingsChange

We don't use this.

GarboMuffin committed:
addons: Fix inaccurate test name

GarboMuffin committed:
addons: Allow editorOnly addons to run in embeds

GarboMuffin committed:
addons: Pull addon changes

clones works in embeds now
onion-skinning supports dark mode

GarboMuffin committed:
addons: Ensure consistent stylesheet order within same addon

GarboMuffin committed:
addons: Pull addon changes

dependabot[bot] committed:
build(deps): bump scratch-paint from `3a46599` to `b99d6e0` (#683)

Bumps [scratch-paint](https://github.com/TurboWarp/scratch-paint) from `3a46599` to `b99d6e0`.
- [Release notes](https://github.com/TurboWarp/scratch-paint/releases)
- [Commits](https://github.com/TurboWarp/scratch-paint/compare/3a46599373eb3418ea164d1ba27221d5e7fd9a16...b99d6e0f23d8e7a1d68d279e6af4c1da693ede31)

---
updated-dependencies:
- dependency-name: scratch-paint
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
addons: Pull addon changes

GarboMuffin committed:
Append dark mode CSS in <head> instead of <body>

Should fix https://github.com/TurboWarp/scratch-gui/issues/685

GarboMuffin committed:
Revert "Append dark mode CSS in <head> instead of <body>"

This reverts commit 7317c83041187027b34bf0fae0c987b21123ca2e.

GarboMuffin committed:
Append addon CSS at the end of <body>

Fixes https://github.com/TurboWarp/scratch-gui/issues/685

dependabot[bot] committed:
build(deps): bump scratch-vm from `e941bb0` to `577217f` (#687)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `e941bb0` to `577217f`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/e941bb0f04c8773c2d6bbff08725371613915159...577217f64f44a3f4cbd74b65065527c1a5659cd6)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Use stream API to save to writable file handles (#686)

* Use stream API to save to writable file handles

* Increase save stream buffer size

Slightly decreases save times

GarboMuffin committed:
Wait for file handle to close before marking saving as finished

GarboMuffin committed:
addons: Pull a find-bar fix

GarboMuffin committed:
Make packager integration less fragile

GarboMuffin committed:
Fix sliders not updating cloud variables

Fixes an upstream bug.

GarboMuffin committed:
addons: Pull upstream changes

GarboMuffin committed:
Update custom extension library item description

GarboMuffin committed:
Add link to extensions.turbowarp.org to the extension library

GarboMuffin committed:
Add tags to extension library (like adacraft)

GarboMuffin committed:
Add stretch, gamepad, and mouse cursor to the extension library

GarboMuffin committed:
Update extension library images

GarboMuffin committed:
Add more Scratch compatibility warnings

GarboMuffin committed:
Update strings

GarboMuffin committed:
Don't load URL extensions again if already loaded

GarboMuffin committed:
Merge pull request #690 from TurboWarp/update-extension-library

Put part of extensions.turbowarp.org into the extension library

GarboMuffin committed:
addons: Pull changes

dependabot[bot] committed:
build(deps): bump scratch-vm from `577217f` to `6049dd5` (#693)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `577217f` to `6049dd5`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/577217f64f44a3f4cbd74b65065527c1a5659cd6...6049dd58fc4e5c4e170ba471be46105173b2c60c)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Fix TurboWarp Blocks image looking wrong in some browsers

GarboMuffin committed:
addons: Port https://github.com/ScratchAddons/ScratchAddons/pull/5592

GarboMuffin committed:
addons: Pull editor-searchable-dropdowns updates

GarboMuffin committed:
addons: Warn when pulling addon with unsafe credit links

GarboMuffin committed:
addons: Import sprite-properties addon

dependabot[bot] committed:
build(deps): bump scratch-vm from `6049dd5` to `20f53fa` (#694)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `6049dd5` to `20f53fa`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/6049dd58fc4e5c4e170ba471be46105173b2c60c...20f53fa4d7be8654ecfc9b095742f183897352cb)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Update addon settings translations

GarboMuffin committed:
New system for TW translations

Instead of storing these in scratch-l10n, we will store these in
scratch-gui as we did a long time ago and currently do for our addon
settings page. This significantly reduces friction involved in updating
translations.

@turbowarp/scratch-l10n will instead be just a slimmed down scratch-l10n

GarboMuffin committed:
Update translations

dependabot[bot] committed:
build(deps): bump @turbowarp/scratch-l10n (#695)

Bumps [@turbowarp/scratch-l10n](https://github.com/TurboWarp/scratch-l10n) from 3.999.20221220224553 to 3.1000.20230110203508.
- [Release notes](https://github.com/TurboWarp/scratch-l10n/releases)
- [Commits](https://github.com/TurboWarp/scratch-l10n/commits)

---
updated-dependencies:
- dependency-name: "@turbowarp/scratch-l10n"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Merge branch 'new-is-old-l10n' into develop

GarboMuffin committed:
addons: Pull sprite-properties changes

GarboMuffin committed:
Update translations

dependabot[bot] committed:
build(deps): bump @turbowarp/scratch-l10n (#697)

Bumps [@turbowarp/scratch-l10n](https://github.com/TurboWarp/scratch-l10n) from 3.1000.20230110203508 to 3.1000.20230112002425.
- [Release notes](https://github.com/TurboWarp/scratch-l10n/releases)
- [Commits](https://github.com/TurboWarp/scratch-l10n/commits)

---
updated-dependencies:
- dependency-name: "@turbowarp/scratch-l10n"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Update translations

GarboMuffin committed:
addons: Add addon to disable cloud variables (marked dangerous)

Closes https://github.com/TurboWarp/scratch-gui/issues/692

GarboMuffin committed:
addons: Add warning when messages in SKIP_MESSAGES do not exist

dependabot[bot] committed:
build(deps): bump scratch-vm from `20f53fa` to `cb2eba9` (#699)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `20f53fa` to `cb2eba9`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/20f53fa4d7be8654ecfc9b095742f183897352cb...cb2eba9eb3b6c568fdbc9cba76747f73b090d287)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump scratch-blocks from `275c0e6` to `3d63bfe` (#698)

Bumps [scratch-blocks](https://github.com/TurboWarp/scratch-blocks) from `275c0e6` to `3d63bfe`.
- [Release notes](https://github.com/TurboWarp/scratch-blocks/releases)
- [Commits](https://github.com/TurboWarp/scratch-blocks/compare/275c0e65da82d47d481e7dc0e8365cf200c436fa...3d63bfe59cf314d838ba8092f71ae881cfb0f7a5)

---
updated-dependencies:
- dependency-name: scratch-blocks
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump scratch-paint from `b99d6e0` to `d84d7a1` (#700)

Bumps [scratch-paint](https://github.com/TurboWarp/scratch-paint) from `b99d6e0` to `d84d7a1`.
- [Release notes](https://github.com/TurboWarp/scratch-paint/releases)
- [Commits](https://github.com/TurboWarp/scratch-paint/compare/b99d6e0f23d8e7a1d68d279e6af4c1da693ede31...d84d7a1f4b1014b8bd86fa5fd2a29bfdc6700a15)

---
updated-dependencies:
- dependency-name: scratch-paint
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Fix ctrl+s to save when caps lock is enabled

GarboMuffin committed:
addons: Fix search-sprites border color in dark mode

dependabot[bot] committed:
build(deps): bump scratch-paint from `d84d7a1` to `a10d2f2` (#704)

Bumps [scratch-paint](https://github.com/TurboWarp/scratch-paint) from `d84d7a1` to `a10d2f2`.
- [Release notes](https://github.com/TurboWarp/scratch-paint/releases)
- [Commits](https://github.com/TurboWarp/scratch-paint/compare/d84d7a1f4b1014b8bd86fa5fd2a29bfdc6700a15...a10d2f2e91992d050f597ceada28437b5728fa15)

---
updated-dependencies:
- dependency-name: scratch-paint
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
addons: Pull changes

Notably, fixes a bug in enhanced fullscreen

GarboMuffin committed:
addons: Fix ommission from previous pull

GarboMuffin committed:
Tune streamed file saving

 - Zip stream allowed to read far ahead of write stream
 - Increased minimum write() buffer size to 1MB
 - Removed transform stream; batching now done in the write stream itself

In all this reduces the time to save Appel from 6.1 to 5.1 seconds on a
computer from 2009 with a spinning hard drive.

Memory usage will increase by a small amount.

GarboMuffin committed:
addons: import paint-snap and some other addon changes

GarboMuffin committed:
addons: Implement more customCssVariables

GarboMuffin committed:
addons: Minor settings page refactor

GarboMuffin committed:
addons: Pull changes

dependabot[bot] committed:
build(deps): bump scratch-vm from `cb2eba9` to `20c6019` (#705)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `cb2eba9` to `20c6019`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/cb2eba9eb3b6c568fdbc9cba76747f73b090d287...20c60193c1c567a65cca87b16d22c51963565a43)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Update translations

GarboMuffin committed:
addons: Pull changes

GarboMuffin committed:
Update translations

WojtekCodesToday commented:
ok

dependabot[bot] committed:
build(deps): bump scratch-vm from `20c6019` to `acd1144` (#710)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `20c6019` to `acd1144`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/20c60193c1c567a65cca87b16d22c51963565a43...acd1144b25fb30ae1ed9a5a813bb7c1e0d57b672)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Implement security manager canFetch, canOpenWindow, canRedirect  (#709)

https://github.com/TurboWarp/extensions/issues/169

GarboMuffin committed:
Update translations

GarboMuffin committed:
Replace translation page link with documentation home link

Translator invitations will be handled in a different way moving forward.
Also updated some old and inaccurate string descriptions.
Closes https://github.com/TurboWarp/scratch-gui/issues/658

GarboMuffin committed:
Update translations

dependabot[bot] committed:
build(deps): bump scratch-vm from `acd1144` to `cd8f28f` (#714)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `acd1144` to `cd8f28f`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/acd1144b25fb30ae1ed9a5a813bb7c1e0d57b672...cd8f28fbba25297a230ca0e304bd0540019cc77f)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Update translations

GarboMuffin committed:
Add GitLab to always trusted fetch origins

Closes https://github.com/TurboWarp/extensions/issues/212

GarboMuffin committed:
Add _ to GitLab Pages regex

WojtekCodesToday commented on src/containers/tw-security-manager.jsx:
```diff
@@ -31,14 +31,20 @@ const fetchOriginsTrustedByUser = new Set();
+    // insensitivity here.
 
     // If we would trust loading an extension from here, we can trust loading resources too.
     isTrustedExtension(parsed.href) ||
```
what is this.

GarboMuffin committed:
Add many origins to the trusted fetch origins list

*.turbowarp.org and *.turbowarp.xyz are now always allowed, so trampoline is always allowed
The APIs of GitHub, GiitLab, Itch, GameJolt, and httpbin are now allowed

Closes https://github.com/TurboWarp/scratch-gui/issues/717

GarboMuffin committed:
Update trusted fetch domain list

Removed regex and added BitBucket Pages

WojtekCodesToday reviewed (commented):
?

dependabot[bot] committed:
build(deps): bump scratch-vm from `cd8f28f` to `c0a1c3d` (#721)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `cd8f28f` to `c0a1c3d`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/cd8f28fbba25297a230ca0e304bd0540019cc77f...c0a1c3db8331f54d5f783f6356df02cd222ef9f8)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump scratch-blocks from `3d63bfe` to `74641be` (#720)

Bumps [scratch-blocks](https://github.com/TurboWarp/scratch-blocks) from `3d63bfe` to `74641be`.
- [Release notes](https://github.com/TurboWarp/scratch-blocks/releases)
- [Commits](https://github.com/TurboWarp/scratch-blocks/compare/3d63bfe59cf314d838ba8092f71ae881cfb0f7a5...74641bedcf1dcea942e7e038c3bec202ad245adb)

---
updated-dependencies:
- dependency-name: scratch-blocks
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Update translations

GarboMuffin committed:
Update extension library images

Most images from https://github.com/TurboWarp/extensions/issues/204
Closes https://github.com/TurboWarp/extensions/issues/259

GarboMuffin committed:
Update extension compatibility message

GarboMuffin committed:
Add files extension to the builtin library

GarboMuffin committed:
Add pointerlock extension to the builtin library

GarboMuffin committed:
Add runtime options extension to builtin library

GarboMuffin committed:
Add utilities extension to builtin library

GarboMuffin committed:
Don't show compatibility warning when clicking custom extension button

GarboMuffin committed:
Move extensions.turbowarp.org link earlier in the library

GarboMuffin committed:
Add delay before enabling extension security prompt buttons

Prevents accidentally closing the prompt immediately.
Closes https://github.com/TurboWarp/scratch-gui/issues/715

GarboMuffin committed:
Update translations

GarboMuffin committed:
addons: Pull folders addon crash fix

GarboMuffin committed:
Update translations

dependabot[bot] committed:
build(deps): bump scratch-vm from `c0a1c3d` to `47adea8` (#724)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `c0a1c3d` to `47adea8`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/c0a1c3db8331f54d5f783f6356df02cd222ef9f8...47adea8107c373e1bc4bc7563493595fe42e432e)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Update translations

dependabot[bot] committed:
build(deps): bump scratch-blocks from `74641be` to `8dea84d` (#725)

Bumps [scratch-blocks](https://github.com/TurboWarp/scratch-blocks) from `74641be` to `8dea84d`.
- [Release notes](https://github.com/TurboWarp/scratch-blocks/releases)
- [Commits](https://github.com/TurboWarp/scratch-blocks/compare/74641bedcf1dcea942e7e038c3bec202ad245adb...8dea84dbb79ab8d51eb5bb27e982283131e0ca52)

---
updated-dependencies:
- dependency-name: scratch-blocks
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump scratch-vm from `47adea8` to `c73f821` (#726)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `47adea8` to `c73f821`.
- [Release notes](https://github.com/TurboWarp/scratch-vm/releases)
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/47adea8107c373e1bc4bc7563493595fe42e432e...c73f821abfeadaa6350d72a28f50587fb0f65674)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump @turbowarp/scratch-l10n (#727)

Bumps [@turbowarp/scratch-l10n](https://github.com/TurboWarp/scratch-l10n) from 3.1000.20230112002425 to 3.1000.20230326022803.
- [Release notes](https://github.com/TurboWarp/scratch-l10n/releases)
- [Commits](https://github.com/TurboWarp/scratch-l10n/commits)

---
updated-dependencies:
- dependency-name: "@turbowarp/scratch-l10n"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

WojtekCodesToday reviewed (commented):
ok

GarboMuffin committed:
Use /api instead of /proxy when accessing trampoline

May help avoid overzealous network filters

WojtekCodesToday commented:
also can someone finally make the editor in production?

the development build is not suitable for some cases.

GarboMuffin committed:
Add ScratchDB to the always-trusted fetch list

https://github.com/TurboWarp/extensions/pull/371

GarboMuffin committed:
Add tw-disable-restore-points addon

https://github.com/TurboWarp/desktop/issues/618

WojtekCodesToday reviewed (approved):
i agree urls

dependabot[bot] committed:
build(deps): bump scratch-render from `5aa2b7e` to `19f0570` (#736)

Bumps [scratch-render](https://github.com/TurboWarp/scratch-render) from `5aa2b7e` to `19f0570`.
- [Commits](https://github.com/TurboWarp/scratch-render/compare/5aa2b7e469ecfe123ead3e3da4555049cdbbbac7...19f057065d99624a0c8b76f4461d08b6341ad630)

---
updated-dependencies:
- dependency-name: scratch-render
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump scratch-vm from `c73f821` to `65bb7b9` (#738)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `c73f821` to `65bb7b9`.
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/c73f821abfeadaa6350d72a28f50587fb0f65674...65bb7b99308c400641bdd0d05b812727e58a7039)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Support loading custom extensions from files and text (#722)

Custom extensions loaded from files and text will have an option to run without the sandbox.
Extensions loaded from files will have their source code stored in the project.
When opened, the security manager will decode the source code and present it to the user (instead of data:...)

GarboMuffin committed:
Update translations

GarboMuffin committed:
Tune new extension-related menus

GarboMuffin committed:
Update extension gallery

GarboMuffin committed:
Update translations

GarboMuffin committed:
Temporarily delist sensing+ from the built in library

https://github.com/TurboWarp/extensions/issues/446

dependabot[bot] committed:
build(deps): bump scratch-render from `19f0570` to `7055489` (#739)

Bumps [scratch-render](https://github.com/TurboWarp/scratch-render) from `19f0570` to `7055489`.
- [Commits](https://github.com/TurboWarp/scratch-render/compare/19f057065d99624a0c8b76f4461d08b6341ad630...705548997ff91a7557f82cbe0a8fe7ac2b1d6a09)

---
updated-dependencies:
- dependency-name: scratch-render
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Tune custom extension modal to help ensure it fits on screen

GarboMuffin committed:
Update code editor button logic for scratch-vm changes

https://github.com/TurboWarp/scratch-vm/pull/137

GarboMuffin committed:
Implement new security manager methods

dependabot[bot] committed:
build(deps): bump scratch-vm from `65bb7b9` to `e387cea` (#741)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `65bb7b9` to `e387cea`.
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/65bb7b99308c400641bdd0d05b812727e58a7039...e387ceadf95925fc28018f6cb6d29ae970192908)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Update translations

dependabot[bot] committed:
build(deps): bump scratch-blocks from `8dea84d` to `f4bb2de` (#742)

Bumps [scratch-blocks](https://github.com/TurboWarp/scratch-blocks) from `8dea84d` to `f4bb2de`.
- [Commits](https://github.com/TurboWarp/scratch-blocks/compare/8dea84dbb79ab8d51eb5bb27e982283131e0ca52...f4bb2de29fbb79cab2e1fd6c4b36c3b442abb31a)

---
updated-dependencies:
- dependency-name: scratch-blocks
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump scratch-vm from `e387cea` to `21915f4` (#743)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `e387cea` to `21915f4`.
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/e387ceadf95925fc28018f6cb6d29ae970192908...21915f4565f08d1dd2b06be1c2393ee1f1e1f485)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump scratch-render from `7055489` to `3352519` (#744)

Bumps [scratch-render](https://github.com/TurboWarp/scratch-render) from `7055489` to `3352519`.
- [Commits](https://github.com/TurboWarp/scratch-render/compare/705548997ff91a7557f82cbe0a8fe7ac2b1d6a09...3352519d2c43ee881ea9dac77fb49a9f462dbee9)

---
updated-dependencies:
- dependency-name: scratch-render
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump scratch-vm from `21915f4` to `f6ab22e` (#745)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `21915f4` to `f6ab22e`.
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/21915f4565f08d1dd2b06be1c2393ee1f1e1f485...f6ab22e280dd09d5080a4684cb83cfd754cb316b)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Include actual error when extension file could not be read

GarboMuffin committed:
Fix typo

GarboMuffin committed:
Update translations

GarboMuffin committed:
Add animated text to the library

dependabot[bot] committed:
build(deps): bump scratch-vm from `f6ab22e` to `19c9e8f` (#747)

Bumps [scratch-vm](https://github.com/TurboWarp/scratch-vm) from `f6ab22e` to `19c9e8f`.
- [Commits](https://github.com/TurboWarp/scratch-vm/compare/f6ab22e280dd09d5080a4684cb83cfd754cb316b...19c9e8f591266ec90b03a24e80c53cbff124164f)

---
updated-dependencies:
- dependency-name: scratch-vm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GarboMuffin committed:
Add "random direction" to "point towards" block

This is a weird one, because the scratch-blocks definition of this block *does* have this,
and it's been supported by scratch-vm forever, they just didn't add it to scratch-gui
